### PR TITLE
Issue #29019 medical borgs on skateboards look weird

### LINKED ISSUE #29019: Medical borgs on skateboards look weird
alternatively: medical cyborgs, when buckled to skateboards which are facing vertically, have their sprite off-centre from the centre of the tile while the board itself is in the centre of the tile, meaning they look weird; they also hover slightly above the board itself

### PR DESCRIPTION
**Pull request by TaeruAlethea**
## About The Pull Request
Makes the cyborgs consistent on scooters and its skateboard subtypes.

## Why It's Good For The Game
Makes the vehicle_specials() proc aware of what the rider is.
Fixes #29019

## Changelog
:cl:
fix: fixed cyborgs using the carbon's offset on scooters and skateboards.
/:cl:

### CODE REVIEW

**Review by ninjanomnom — APPROVED** [review]

File: code/datums/components/riding/riding_vehicle.dm [hunk 1]
```diff
@@ -135,9 +135,12 @@
 /datum/component/riding/vehicle/scooter
 	ride_check_flags = RIDER_NEEDS_LEGS | RIDER_NEEDS_ARMS | UNBUCKLE_DISABLED_RIDER
 
-/datum/component/riding/vehicle/scooter/handle_specials()
+/datum/component/riding/vehicle/scooter/handle_specials(mob/living/riding_mob)
 	. = ..()
-	set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0), TEXT_SOUTH = list(-2), TEXT_EAST = list(0), TEXT_WEST = list( 2)))
+	if(iscyborg(riding_mob))
+		set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0), TEXT_SOUTH = list(0), TEXT_EAST = list(0), TEXT_WEST = list( 2)))
```
Comment (dragomagol) [inline code/datums/components/riding/riding_vehicle.dm:141]:
```suggestion
		set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0), TEXT_SOUTH = list(0), TEXT_EAST = list(0), TEXT_WEST = list(2)))
```

File: code/datums/components/riding/riding_vehicle.dm [hunk 2]
```diff
@@ -135,9 +135,12 @@
...
+	else
+		set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list( 2), TEXT_SOUTH = list(-2), TEXT_EAST = list(0), TEXT_WEST = list( 2)))
```
Comment (dragomagol) [inline code/datums/components/riding/riding_vehicle.dm:143]:
```suggestion
		set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(2), TEXT_SOUTH = list(-2), TEXT_EAST = list(0), TEXT_WEST = list(2)))
```

File: code/datums/components/riding/riding_vehicle.dm (hunk 1 above)
Comment (dragomagol) [reply code/datums/components/riding/riding_vehicle.dm:141]: Couple of extra spaces here

File: code/datums/components/riding/riding_vehicle.dm (hunk 2 above)
Comment (dragomagol) [reply code/datums/components/riding/riding_vehicle.dm:143]: And here

**Comment by Funce** [comment]
Adding the line `Fixes #<issuenumber>` in your description links the issue to the PR.
For other forms you can see: 
https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

**Pull request merged**